Enable GitHub Actions

MoshiBin opened the pull request:
- **Linters** currently includes only Ruff,  but can be easily amended to add linters as needed (maybe mypy, markdown lint, etc).
- **Test** performs a basic installation of the package and tries to run the CLI with `--help`. It may be extended in the future to run extra tests as needed. Note that this action currently fails because it tries to test Python 3.12, which requires #18 to be merged.

AndBondStyle commented on .github/workflows/linters.yml:
```diff
@@ -0,0 +1,8 @@
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: chartboost/ruff-action@v1
```
Let's use pre-commit action instead: https://github.com/pre-commit/action (ruff is already included)

MoshiBin replied:
Done

AndBondStyle commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,23 @@
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.11", "3.12"]
```
Let's also include older versions maybe? From 3.8 to 3.12 for example

AndBondStyle commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,23 @@
+      - name: Set up Poetry
+        uses: snok/install-poetry@v1
+
+      - run: poetry install --no-interaction
```
```suggestion
      # todo: add venv caching
      - run: poetry install --no-interaction
```

MoshiBin commented:
Added support & testing for Python 3.10 (which removes the dependency on #18), replaced the ruff action with pre-commit, added the `todo` notice for caching poetry venvs - I'll look into this later.
